app/controllers/favorites_controller.rb

```diff
@@ -14,14 +14,13 @@ def update
 
   def destroy
     pet = Pet.find(params[:pet_id])
-    favorites.delete_pet(pet.id)
-    session[:favorites] = favorites.contents
+    session[:favorites].delete(params[:pet_id])
     flash[:notice] = "#{pet.name} has been removed from your favorites list."
     redirect_back(fallback_location: root_path)
   end
 
   def destroy_all
-    favorites.reset
+    session[:favorites] = {}
```

Comment:
Same here -- this will work much better than just resetting the hash in the PORO.